mutate/mutate.go

```diff
@@ -298,6 +302,27 @@ func (m *Mutator) Add(ctx context.Context, mediaType string, r io.Reader, histor
 	return desc, nil
 }
 
+// AddExisting adds a blob that already exists to the layer, using the user
+// specified DiffID. It currently checks that the layer exists, but does not
+// validate the DiffID.
+func (m *Mutator) AddExisting(ctx context.Context, desc ispec.Descriptor, history *ispec.History, diffID digest.Digest) error {
+	if err := m.cache(ctx); err != nil {
+		return errors.Wrap(err, "getting cache failed")
+	}
+
+	// make sure the blob exists
+	blob, err := m.engine.GetBlob(ctx, desc.Digest)
+	if err != nil {
+		return err
+	}
+	blob.Close()
+
+	// perhaps we should optionally verify the blob's diffid against its compression?
```

Comment:
If we're going to have an API that computes the DiffID, it's probably better to just make that API generate teh DiffID and use it rather than comparing against a value the caller provides -- but that'd need to be a separate API to this more "light-weight" version where you just add a layer without reading or writing any layer blobs (once I fix the GetBlob thing.)

Reply:
Yeah, and I really don't want to compute the DiffID since it will be expensive in most cases.